nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -439,6 +439,31 @@ public void testForMethodReferenceInAnAssignment() {
         .doTest();
   }
 
+  @Test
+  public void testForMethodReferenceReturnType() {
```

Comment:
As noted in an offline discussion, this test looks good but let's add tests for more (pseudo-)assignments, like passing a method reference as a function parameter and returning a method reference